Tests/Approval.Tests/ApprovedApi/FluentAssertions/net47.verified.txt

```diff
@@ -1119,76 +1120,63 @@ namespace FluentAssertions.Events
     {
         public AssertionScope() { }
         public AssertionScope(FluentAssertions.Execution.IAssertionStrategy assertionStrategy) { }
-        public AssertionScope(System.Lazy<string> context) { }
-        public AssertionScope(string context) { }
-        public string CallerIdentity { get; }
-        public System.Lazy<string> Context { get; set; }
+        public AssertionScope(System.Func<string> name) { }
+        public AssertionScope(string name) { }
         public FluentAssertions.Formatting.FormattingOptions FormattingOptions { get; }
-        public FluentAssertions.Execution.AssertionScope UsingLineBreaks { get; }
+        public System.Func<string> Name { get; }
         public static FluentAssertions.Execution.AssertionScope Current { get; }
-        public void AddNonReportable(string key, object value) { }
         public void AddPreFormattedFailure(string formattedFailureMessage) { }
-        public void AddReportable(string key, System.Func<string> valueFunc) { }
-        public void AddReportable(string key, string value) { }
         public void AppendTracing(string tracingBlock) { }
-        public void AssumeSingleCaller() { }
-        public FluentAssertions.Execution.AssertionScope BecauseOf(FluentAssertions.Execution.Reason reason) { }
-        public FluentAssertions.Execution.AssertionScope BecauseOf(string because, params object[] becauseArgs) { }
-        public FluentAssertions.Execution.Continuation ClearExpectation() { }
         public string[] Discard() { }
         public void Dispose() { }
-        public FluentAssertions.Execution.Continuation FailWith(System.Func<FluentAssertions.Execution.FailReason> failReasonFunc) { }
-        public FluentAssertions.Execution.Continuation FailWith(string message) { }
-        public FluentAssertions.Execution.Continuation FailWith(string message, params System.Func<object>[] argProviders) { }
-        public FluentAssertions.Execution.Continuation FailWith(string message, params object[] args) { }
-        public FluentAssertions.Execution.AssertionScope ForCondition(bool condition) { }
-        public FluentAssertions.Execution.AssertionScope ForConstraint(FluentAssertions.OccurrenceConstraint constraint, int actualOccurrences) { }
-        public T Get<T>(string key) { }
-        public FluentAssertions.Execution.GivenSelector<T> Given<T>(System.Func<T> selector) { }
         public bool HasFailures() { }
-        public FluentAssertions.Execution.AssertionScope WithDefaultIdentifier(string identifier) { }
-        public FluentAssertions.Execution.AssertionScope WithExpectation(string message, params object[] args) { }
     }
     public class Continuation
     {
-        public FluentAssertions.Execution.IAssertionScope Then { get; }
-        public static bool op_Implicit(FluentAssertions.Execution.Continuation continuation) { }
+        public FluentAssertions.Execution.AssertionChain Then { get; }
```

Comment:
Did you check whether creating an interface `IAssertionChain` would be helpful for the FA unit tests itself (now and in future)?

Reply:
Why would we need it?

Reply:
For the same reason I added to `AssertionChain`.
Interfaces allows mocking in unit test, I'm sure you know this.
So, according to my experience nearly everything provided to public should be an interface instead of a class to be mockable.
The old version of this property returned an interface too. Why did you decide in the past to provide an interface, but not today?

Reply:
Why on earth would you ever want to mock an assertion chain? Can you give me an example?

Reply:
I do not have a real example. I analyzed just with my theoretical eye.
I think, public API should use interfaces as much as possible.

When I'm not able to create my very own, non-static instance of `AssertionChain`  then I would like to have an instance where `AssertionChain` is used.

We can postpone this to the time when such change is required for a specific test. (I do not think of a user test but on an FA internal test.)

Reply:
> I think, public API should use interfaces as much as possible.

I do not share that opinion unfortunately.

Reply:
> I think, public API should use interfaces as much as possible.

IMHO this applies to projects like https://github.com/quartznet/quartznet where the only task of the "library" is to provide a "shape" and let the user do the whole implementation work.
This doesn't fully apply to FA (apart from the whole custom `Formatters` and `IEquivalencyStep`s).